fastly/domain_inspector.go

```diff
@@ -0,0 +1,181 @@
+	Status1xx                  uint64  `mapstructure:"status_1xx"`
+	Status200                  uint64  `mapstructure:"status_200"`
+	Status204                  uint64  `mapstructure:"status_204"`
+	Status206                  uint64  `mapstructure:"status_206"`
+	Status2xx                  uint64  `mapstructure:"status_2xx"`
+	Status301                  uint64  `mapstructure:"status_301"`
+	Status302                  uint64  `mapstructure:"status_302"`
+	Status304                  uint64  `mapstructure:"status_304"`
+	Status3xx                  uint64  `mapstructure:"status_3xx"`
+	Status400                  uint64  `mapstructure:"status_400"`
+	Status401                  uint64  `mapstructure:"status_401"`
+	Status403                  uint64  `mapstructure:"status_403"`
+	Status404                  uint64  `mapstructure:"status_404"`
+	Status416                  uint64  `mapstructure:"status_416"`
+	Status429                  uint64  `mapstructure:"status_429"`
+	Status4xx                  uint64  `mapstructure:"status_4xx"`
+	Status500                  uint64  `mapstructure:"status_500"`
+	Status501                  uint64  `mapstructure:"status_501"`
+	Status502                  uint64  `mapstructure:"status_502"`
+	Status503                  uint64  `mapstructure:"status_503"`
+	Status504                  uint64  `mapstructure:"status_504"`
+	Status505                  uint64  `mapstructure:"status_505"`
+	Status5xx                  uint64  `mapstructure:"status_5xx"`
+	Timestamp                  uint64  `mapstructure:"timestamp"`
+}
+
+// DomainMeta is the meta section returned for /metrics/domain responses
+type DomainMeta struct {
+	Downsample string            `mapstructure:"downsample"`
+	End        string            `mapstructure:"end"`
+	Filters    map[string]string `mapstructure:"filters"`
+	GroupBy    string            `mapstructure:"group_by"`
+	Limit      int               `mapstructure:"limit"`
+	Metric     string            `mapstructure:"metric"`
+	NextCursor string            `mapstructure:"next_cursor"`
+	Sort       string            `mapstructure:"sort"`
+	Start      string            `mapstructure:"start"`
+}
+
+// GetDomainMetricsInput is the input to a DomainMetrics request.
+type GetDomainMetricsInput struct {
+	// Cursor is the value from a previous response to retrieve the next page. To request the first page, this should be empty.
+	Cursor string
+	// Datacenters limits query to one or more specific POPs.
+	Datacenters []string
+	// Domains limit query to one or more specific domains.
+	Domains []string
+	// Downsample is the duration of sample windows.
+	Downsample string
+	// End is a valid ISO-8601-formatted date and time, or UNIX timestamp, indicating the exclusive end of the query time range. If not provided, a default is chosen based on the provided downsample value.
+	End time.Time
+	// GroupBy is the dimensions to return in the query.
+	GroupBy []string
```

Comment:
I don't see this being referenced or passed through anywhere.

Reply:
Thanks for catching. Fixed.